common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -147,6 +147,27 @@ def test_provider_disabled_for_mismatching_site(self):
         prov = self.configure_google_provider(visible=True, enabled=True, site=site_b)
         self.assertEqual(prov.enabled_for_current_site, False)
 
+    @with_site_configuration(SITE_DOMAIN_A)
+    def test_providers_with_same_key_independent_across_sites(self):
```

Comment:
I believe we can have another test case where each `site` will have both oauth_provider set and SAML_provider set. Then the return should be the latest oauth_provider and latest saml provider in a list.

Reply:
Good idea.